salt_stack/salt/boss-tools/files/firstboot.py

```diff
@@ -97,8 +97,10 @@ def set_hostname():
         bossutils.configuration.download_and_save()
         #read_vault_token() # Not currently supported when generating access tokens
         set_hostname()
+    except Exception as ex:
+        logging.exception('Problem during initial configuration: {}'.format(ex))
     except:
-        log.exception("Problem during initial configuration")
+        logging.exception("Problem during initial configuration")
```

Comment:
Why this second exception block? The only time this may be [used](https://docs.python.org/3/library/exceptions.html#Exception) is if a system-existing exception is raised (Like `KeyboardInterrupt` from `CTRL-C`)

Reply:
You originally had just `except:` so I left it in, in case you had a reason for it. :)

Reply:
But as I pointed out, you basically removed the plain `except:` block from ever being executed with the new `except Exception ...` block.

Reply:
Ok, makes sense to remove since we can't log any additional info because it's not possible to do `except as err:`, right?

Reply:
You cannot log any additional information because it will never be executed. The `except Exception as ex:` block covers almost all Exceptions that can be raised and all of the ones we care about.